src/fsharp/FSharp.Core/map.fs

```diff
@@ -598,6 +613,56 @@ namespace Microsoft.FSharp.Collections
             member s.TryGetValue(k,r) = if s.ContainsKey(k) then (r <- s.[k]; true) else false
             member s.Remove(k : 'Key) = ignore(k); (raise (NotSupportedException(SR.GetString(SR.mapCannotBeMutated))) : bool)
 
+        interface System.Collections.IDictionary with
+            member m.Item
+                with get key =
+                    match key with
+                    | null ->
+                        // According to the documentation for IDictionary, implementations should
+                        // raise ArgumentNullException if the key passed to the indexer is null.
+                        nullArg "key"
+                    | :? 'Key as key ->
+                        match m.TryFind key with
+                        | Some v -> box v
+                        | None -> null
+                    | _ -> null
+                and  set key value =
```

Comment:
@KevinRansom I agree, the partial contract implementations aren't ideal; in this case though, we're just adding (for the most part) the non-generic versions of interfaces already implemented by ``Set<_>`` and ``Map<_,_>``.

@eiriktsarpalis gave some good reasons to justify implementing the non-generic versions of these interfaces in his suggestion (fsharp/fslang-suggestions#473).